File: src/clients/smar_bank_app/lib/models/endereco.dart
class Endereco {
  String cep;
  String logradouro;
  String complemento;
  String numero;
  String bairro;
  String cidade;
  int codigoIBGE;

  Endereco(
      {this.cep,
        this.logradouro,
        this.complemento,
        this.numero,
        this.bairro,
        this.cidade,
        this.codigoIBGE});

  Endereco.fromJson(Map<String, dynamic> json) {
    cep = json['cep'];
    logradouro = json['logradouro'];
    complemento = json['complemento'];
    numero = json['numero'];
    bairro = json['bairro'];
    cidade = json['cidade'];
    codigoIBGE = json['codigoIBGE'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['cep'] = this.cep;
    data['logradouro'] = this.logradouro;
    data['complemento'] = this.complemento;
    data['numero'] = this.numero;
    data['bairro'] = this.bairro;
    data['cidade'] = this.cidade;
    data['codigoIBGE'] = this.codigoIBGE;
    return data;
  }
}

File: src/services/SmartBank/SmartBankUsers.Application/ViewModels/DataValid/Endereco.cs
using Newtonsoft.Json;

namespace SmartBank.Application.ViewModels.DataValid
{
    public class Endereco
    {
        public Endereco(string logradouro, string numero, string complemento, string bairro, string cep, string municipio, string uf)
        {
            Logradouro = logradouro;
            Numero = numero;
            Complemento = complemento;
            Bairro = bairro;
            Cep = cep;
            Municipio = municipio;
            Uf = uf;
        }

        [JsonProperty("logradouro")]
        public string Logradouro { get; set; }

        [JsonProperty("numero")]
        public string Numero { get; set; }

        [JsonProperty("complemento")]
        public string Complemento { get; set; }

        [JsonProperty("bairro")]
        public string Bairro { get; set; }

        [JsonProperty("cep")]
        public string Cep { get; set; }

        [JsonProperty("municipio")]
        public string Municipio { get; set; }

        [JsonProperty("uf")]
        public string Uf { get; set; }
    }
}
